iOS/ViewController.swift

```diff
@@ -8,18 +8,70 @@
 
 import UIKit
 
-class ViewController: UIViewController {
-
+class ViewController: UIViewController, SpeakerDelegate, UIPickerViewDelegate, UIPickerViewDataSource {
+    
+    @IBOutlet var textview: UITextView!
+    var languagePicker: UIPickerView = UIPickerView()
+    let speaker = Speaker.defaultSpeaker
+    var languageNames = [String]()
+    
+    @IBAction func playClicked(_ sender: AnyObject) {
+        if let text = self.textview.text {
+            self.speaker.speakText(text: text)
+        }
+    }
+    @IBAction func changeVoiceClicked(_ sender: AnyObject) {
+        // create action sheet dynamically
+        let alertView = UIAlertController(title: "Select Launguage", message: "\n\n\n\n\n\n\n\n\n\n", preferredStyle: UIAlertControllerStyle.actionSheet);
+        self.languagePicker.center.x = self.view.center.x
+        let action = UIAlertAction(title: "DONE", style: .default) { (action) in
+            let value = self.languagePicker.selectedRow(inComponent: 0)
+            self.speaker.changeLanguage(language: self.speaker.voices[value].language)
+        }
+        alertView.addAction(action)
+        alertView.view.addSubview(languagePicker)
+        present(alertView, animated: true, completion: nil)
+        self.languagePicker.autoresizingMask = [UIViewAutoresizing.flexibleLeftMargin, UIViewAutoresizing.flexibleRightMargin]
+    }
+    
+    func speaker(speaker: Speaker, didFinishSpeechString: String) {
+        print("didFinishSpeechString : ", didFinishSpeechString)
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        
+        // create popup picker view
+        languagePicker = UIPickerView(frame: CGRect(x: 0, y: 20, width: 300, height: 250))
```

Comment:
I think the size must not be set as fixed, but flexible ralative to the alert view width.
for example:
width = alertView's width - 40